app/Http/Controllers/ProfileController.php

```diff
@@ -72,6 +73,23 @@ public function getPerformance(Request $request)
         return $this->jsonSuccess($performance->aggregateForTimeRange($profile, $startDate, $endDate));
     }
 
+    public function getPerformancePerTask(Request $request)
+    {
+        GenericModel::setCollection('tasks');
+        $task = GenericModel::where('owner', '=', $request->route('id'))->first();
```

Comment:
This is supposed to load task by ID, not first task for that user. This implementation will return random data as is now - please update to load by task ID - `GenericModel::find($taskId)`